test/stmt/foreach.swift

```diff
@@ -17,7 +17,7 @@ func bad_containers_2(bc: BadContainer2) {
 }
 
 struct BadContainer3 : Sequence { // expected-error{{type 'BadContainer3' does not conform to protocol 'Sequence'}}
-  func makeIterator() { } // expected-note{{candidate can not infer 'Iterator' = '()' because '()' is not a nominal type and so can't conform to 'IteratorProtocol'}}
+  func makeIterator() { } // expected-note{{candidate can not infer 'Iterator' = '()' because '()' is not a struct, class or enum type and so can't conform to 'IteratorProtocol'}}
```

Comment:
Here, too, seems weird to me that it's describing what `()` is _not_. Perhaps just "...because tuple type '()' can't conform to 'IteratorProtocol'."